### PR TITLE
Bytecode support for HTTP requests

### PR DESCRIPTION
**Pull request by vkagamlyk**
Added Gremlin bytecode support for HTTP requests for Java and Python GLV's without translation to Gremlin script.
Added serialization for entire RequestMessage.
Payload type distinguished by Content-Type header .

### CODE REVIEW

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java [hunk 1]
```diff
@@ -70,10 +76,55 @@ public class HttpHandlerUtil {
...
+     *     Request body contains serialized RequestMessage
      */
-    public static RequestMessage getRequestMessageFromHttpRequest(final FullHttpRequest request) {
-        final String contentType = Optional.ofNullable(request.headers().get(HttpHeaderNames.CONTENT_TYPE)).orElse("application/json");
+    public static RequestMessage getRequestMessageFromHttpRequest(final FullHttpRequest request,
+                                                                  Map<String, MessageSerializer<?>> serializers) throws SerializationException {
+        final String contentType = request.headers().get(HttpHeaderNames.CONTENT_TYPE);
+
+        if (request.method() == POST && contentType != null && !contentType.equals("application/json") && serializers.containsKey(contentType)) {
+            final MessageSerializer<?> serializer = serializers.get(contentType);
```
Comment (kenhuuu) [inline gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java:100]: What happens if the requested serializer doesn't exist so this value is null?

File: gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/HttpDriverIntegrateTest.java [hunk 2]
```diff
@@ -97,7 +121,8 @@ public void shouldSubmitBytecodeWithGraphBinary() throws Exception {
                 .create();
         try {
             final GraphTraversalSource g = traversal().withRemote(DriverRemoteConnection.using(cluster));
-            assertEquals("2", g.inject("2").toList().get(0));
+            final String result = g.inject("2").toList().get(0);
```
Comment (kenhuuu) [inline gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/HttpDriverIntegrateTest.java:124]: Why was this change necessary?

Comment (vkagamlyk) [reply gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/HttpDriverIntegrateTest.java:124]: not necessary, consequences of debugging

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java (hunk 1 above)
Comment (vkagamlyk) [reply gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java:100]: 1. `serializers.containsKey(contentType)`, so serializer is present
2. null value should not be in `serilizers` because https://github.com/apache/tinkerpop/blob/8c755b077369a327c2bd7a74fd33cd15073cf92a/gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/AbstractChannelizer.java#L278

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java [hunk 3]
```diff
@@ -70,10 +76,55 @@ public class HttpHandlerUtil {
...
+            final int first = buffer.readByte();
+            // payload can be plain json or can start with additional header with content type.
+            // if first character is not "{" (0x7b) then need to verify is correct serializer selected.
+            if (first != 0x7b) {
+                final byte[] bytes = new byte[first];
+                buffer.readBytes(bytes);
+                final String mimeType = new String(bytes, StandardCharsets.UTF_8);
+
+                if (Arrays.stream(serializer.mimeTypesSupported()).noneMatch(t -> t.equals(mimeType)))
+                    throw new IllegalArgumentException("Mime type mismatch. Value in content-type header is not equal payload header.");
```
Comment (kenhuuu) [inline gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java:114]: You should check to see how the GLVs will handle this. Most of the time they expect a fully formed RequestMessage in order to properly map the error to the original request.

File: gremlin-server/conf/gremlin-server-rest-modern.yaml [hunk 4]
```diff
@@ -28,7 +28,11 @@ scriptEngines: {
                org.apache.tinkerpop.gremlin.jsr223.ImportGremlinPlugin: {classImports: [java.lang.Math], methodImports: [java.lang.Math#*]},
                org.apache.tinkerpop.gremlin.jsr223.ScriptFileGremlinPlugin: {files: [scripts/generate-modern.groovy]}}}}
 serializers:
-  - { className: org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0, config: { ioRegistries: [org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerIoRegistryV3d0] }}         # application/json
+  - { className: org.apache.tinkerpop.gremlin.util.ser.GraphSONUntypedMessageSerializerV3, config: { ioRegistries: [org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerIoRegistryV3] }}     # application/vnd.gremlin-v3.0+json;types=false
```
Comment (kenhuuu) [inline gremlin-server/conf/gremlin-server-rest-modern.yaml:31]: I think these are in the wrong package for 3.6-dev.

File: gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java (hunk 3 above)
Comment (vkagamlyk) [reply gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/HttpHandlerUtil.java:114]: `HttpGremlinEndpointHandler` should catch exception and send RequestMessage to GLV's 
Lines 151-152

**Comment by Cole-Greer** [comment]
LGTM, VOTE +1

**Comment by kenhuuu** [comment]
VOTE +1

**Comment by vkagamlyk** [comment]
VOTE+1
